jetty-websocket/jetty-websocket-api/src/main/java/org/eclipse/jetty/websocket/api/RemoteEndpoint.java

```diff
@@ -36,35 +37,89 @@
     void sendBytes(ByteBuffer data) throws IOException;
 
     /**
-     * Initiates the asynchronous transmission of a binary message. This method returns before the message is
-     * transmitted. Developers may use the returned
-     * Future object to track progress of the transmission.
+     * Initiates the asynchronous transmission of a binary message. This method returns before the message is transmitted.
+     * Developers may provide a callback to be notified when the message has been transmitted or resulted in an error.
+     *
+     * @param data     the data being sent
+     * @param callback callback to notify of success or failure of the write operation
+     */
+    void sendBytes(ByteBuffer data, WriteCallback callback);
+
+    /**
+     * Initiates the asynchronous transmission of a binary message. This method returns before the message is transmitted.
+     * Developers may use the returned Future object to track progress of the transmission.
      *
      * @param data the data being sent
      * @return the Future object representing the send operation.
      */
     Future<Void> sendBytesByFuture(ByteBuffer data);
```

Comment:
So can we deprecate `sendBytesByFuture` as using a callback is a much better way to be async.... or at the least can we return completableFuture?